parallelize_tests.go

```diff
@@ -9,19 +9,44 @@ package main
 import (
 	"context"
 	"errors"
+	"io/ioutil"
+	"log"
+	"math"
 	"math/rand"
 	"os"
 	"os/exec"
 	"os/signal"
 	"runtime"
+	"strconv"
 	"strings"
 	"sync"
```

Comment:
Would you mind rewriting this script in Python? It will make this repository easier to maintain for other folks  who may not be familiar with Go.